hail/src/main/scala/is/hail/utils/TextTableReader.scala

```diff
@@ -16,27 +16,33 @@ import org.apache.spark.rdd.RDD
 
 import scala.util.matching.Regex
 
+abstract class TextReaderOptions {
+  val comment: Array[String]
+  val hasHeader: Boolean
+
+  private lazy val commentStartsWith: Array[String] = comment.filter(_.length == 1)
+  private lazy val commentRegexes: Array[Regex] = comment.filter(_.length > 1).map(_.r)
+
+  final def isComment(line: String): Boolean =
+    commentStartsWith.exists(pattern => line.startsWith(pattern)) || commentRegexes.exists(pattern => pattern.matches(line))
+}
+
 case class TextTableReaderOptions(
   files: Array[String],
   typeMapStr: Map[String, String],
   comment: Array[String],
   separator: String,
   missing: Set[String],
-  noHeader: Boolean,
+  hasHeader: Boolean,
```

Comment:
There are a few more places in this class that use noHeader:

https://github.com/hail-is/hail/blob/45d42256297dcdd20fc89671fbcb74c82647e56d/hail/src/main/scala/is/hail/utils/TextTableReader.scala#L223
 -> hasHeader

https://github.com/hail-is/hail/blob/45d42256297dcdd20fc89671fbcb74c82647e56d/hail/src/main/scala/is/hail/utils/TextTableReader.scala#L256

https://github.com/hail-is/hail/blob/45d42256297dcdd20fc89671fbcb74c82647e56d/hail/src/main/scala/is/hail/utils/TextTableReader.scala#L272

the last 2 !hasHeader

Reply:
thanks! That's why this was failing...